pallets/nft/src/mock.rs

```diff
@@ -121,10 +122,37 @@ pub fn run_to_block(n: u32) {
 	}
 }
 
-pub fn events() -> Vec<Event> {
-	let evt = System::events().into_iter().map(|evt| evt.event).collect::<Vec<_>>();
+pub fn events_filter<T: 'static>() -> Vec<Event> {
+	let mut evt = System::events();
 
-	System::reset_events();
+	evt.retain(|evt| if_right_events::<T>(&evt.event));
+	return evt.into_iter().map(|evt| evt.event).collect::<Vec<_>>();
+}
+
+pub fn if_right_events<T: 'static>(evt: &Event) -> bool {
+	if TypeId::of::<T>() == TypeId::of::<Event>() {
+		return true;
+	} else {
+		match evt {
+			Event::System(i) => return if_right_raw_events::<T>(i),
+			Event::Balances(i) => return if_right_raw_events::<T>(i),
+			Event::Nft(i) => return if_right_raw_events::<T>(i),
+		}
+	}
+}
 
-	evt
+pub fn if_right_raw_events<T: 'static>(s: &dyn Any) -> bool {
+	if let Some(_) = s.downcast_ref::<T>() {
+		true
+	} else {
+		false
+	}
+}
+
+pub fn get_vector<T>(vector: &Vec<T>, index: isize) -> &T {
```

Comment:
Why we need this? If you have a lot of events, you still have no idea the index. If you just have a few, this makes code longer and does not provide great convenience

Reply:
The implementation of this allows input like -1,-2 to enable get the "latest" element in the vector rather then manually tracing a positive index number accordingly. At least I did not find too many direct method in Vec class (I try use .last() method, but something is not implemented and compiler fails).

I am a little confused about the index requirement.